New features

nullzone opened the pull request:
- Support for new optional parameters in the database connectors:
  - auth_user
  - auth_pass
  - pool_size
  
  Those parameters are obviously integrated automatically in the hiera definition, e.g:
  
  yourmodule_db::pgbouncer_dbs      : 
  - source_db         : 'logdb'
    host              : '127.0.0.1'
    dest_db           : 'mydb'
    auth_user         : 'myrole'
    auth_pass         : 'mypassword'
    pool_size         : '15'
- Support for the [users] section available after pgbouncer 1.6.1 which allow to define a different pool_mode per user.
  It can be dine using the new optional parameter to be used in the existing userlist hiera definition. E.g.:
  
  yourmodule_db::pgbouncer_userlist : 
  - user              : 'pguser1'
    password          : 'pgpass1'
  - user              : 'pguser2'
    password          : 'pgpass2'
    pool_mode         : 'session'

nullzone committed:
Update pgbouncer.ini.databases.part2.erb

nullzone committed:
pool options

nullzone committed:
I-d go to sleep...

nullzone committed:
Update pgbouncer.ini.databases.part2.erb

nullzone committed:
Update CHANGELOG.md

nullzone committed:
Update init.pp

nullzone committed:
Update metadata.json

nullzone committed:
[users] section in the ini file to give a per user pool_mode support

nullzone committed:
Update userlist.pp

nullzone committed:
Update userlist.pp

nullzone committed:
Update pgbouncer.ini.users.part.erb

nullzone committed:
Update userlist.pp

nullzone committed:
Update init.pp

nullzone committed:
Update pgbouncer.ini.users.part.erb

nullzone committed:
Update userlist.pp

nullzone commented:
Some changes.

cjestel commented:
Cool! I had this on my list of things to figure out. I should be able to test this later this week or early next and will merge and get a new version out as soon as I'm done testing.

nullzone committed:
port support

nullzone committed:
default values

default values should be consistent for all pgbouncer versions.

if pool_mode is transaction, server_reset_query should be empty. Let's keep server_reset_query with a good value and change the pool_mode to session.

nullzone committed:
Update pgbouncer.ini.databases.part2.erb

nullzone commented:
Hello there,

are you thinking about merging these changes or should I think about keeping a permanent fork?

Cheers,

Alberto

adamcstephens commented:
@nullzone I'll get with @cjestel when he returns next week and review this. There should be no need for a permanent fork.

cjestel commented:
Sorry for the delay in getting this merged. Change looks good and I'll get a new release cut Monday when I'm back in the office.

cjestel commented on manifests/userlist.pp:
```diff
@@ -13,4 +14,11 @@
     order   => '01',
   }
+
+  concat::fragment { "${paramtmpfile}_users":
```
I missed this in my local testing prior to merge. This works if the userlist is only called once, but is duplicated and causes compilation issues if it is called a second time. I'm going to re-work this (and the supporting files) to allow multiple calls to the defined type. My initial thoughts are tying in the username like what happens above, and then changing up the template to have an additional piece so it doesn't re-declare the `[users]` section of the config. I wanted to give you a heads up that this will be changing so you know to pull and perform any re-tests as necessary with the new release. I'm working through the fix today and will post the new release as soon as possible (should be this week).
